flexmeasures/ui/templates/crud/asset.html

```diff
@@ -369,6 +483,436 @@ <h3>All child assets for {{ asset.name }}</h3>
+                <div class="card m-0">
+                    <div class="card-body p-0 sensor-card">
+                        <h5 class="card-title">${sensor.name}</h5>
+                        <p class="card-text">
+                            <b>ID:</b> <a href="${apiBasePath}/sensors/${sensor.id}">${sensor.id}</a>, 
+                            <b>Unit:</b> ${sensor.unit},
+                            <b>Asset:</b> ${Asset.name},
+                            <b>Account:</b> ${Account?.name ? Account.name : "PUBLIC"}
+                        </p>
+                        ${graphIndex !== undefined 
+                        ? `<button class="btn btn-primary btn-sm" onclick="addSensorToExistingGraph(${graphIndex}, ${sensor.id})">Add to '${selectedGraphTitle}' Graph</button>` 
+                        : `<button class="btn btn-primary btn-sm" onclick="addSensorAsGraph(${sensor.id})">Add to Graph</button>`
+                        }
+                        
+                    </div>
+                </div>
+            `;
+
+            apiSensorsList.appendChild(col);
+        });
+    }
+
+    async function updateAssetSensorsToShow() {
+        const graphSpinnerElement = document.getElementById('spinner');
+        // Show spinner
+        graphSpinnerElement.style.display = 'block';
+
+        const apiURL = apiBasePath + "/api/v3_0/assets/{{ asset.id }}";
+        const dataJson = JSON.stringify( {sensors_to_show: JSON.stringify(data)} );
+    
+        const response = await fetch(apiURL, {
+            method: "PATCH",
+            headers: {
+                "Content-Type": "application/json",
+            },
+            body: dataJson,
+        });
+    
+        if (!response.ok) {
+            showToast("Failed to update the asset", "error");
+        } else{
+            document.dispatchEvent(new Event('sensorsToShowUpdated'));
+            showToast("Changes saved successfully", "success");
+        }
+    }
+
+    // Add a sensor as a new graph card
+    function addSensorAsGraph(id) {
+        const newAsset = {
+            title: "No Title", 
+            sensors: [id], 
+        };
+        data.push(newAsset);
+        renderGraphCards();
+    }
+
+    // Add blank graph to the graph cards
+    function addBlankGraph() {
+        const newAsset = {
+            title: "No Title " + (data.length + 1), 
+            sensors: [], 
+        };
+        selectedGraphTitle = newAsset.title;
+        data.push(newAsset);
+        selectGraph(data.length - 1)
+        renderGraphCards();
+    }
+
+    // Add Sensor to an existing graph card
+    function addSensorToExistingGraph(graphIndex, sensorId) {
+        data[graphIndex].sensors.push(sensorId);
+        renderGraphCards();
+    }
+
+    // Remove sensor from the graph sensor list
+    function removeSensorFromGraph(graphIndex, sensorIndex) {
+        data[graphIndex].sensors.splice(sensorIndex, 1);
+        selectedGraphTitle = data[graphIndex].title;
```

Comment:
Why do you re-set the graph title here? Seems unnecessary.

Reply:
This is actually for the rendered sensors not to show "undefined" cause the graph they were pointing to was removed. But this logic is outdated and there is now a bug with this action.

I have updated the code with a fix to properly handle the state management here